ifunc attribute not detected on clang++

**Describe the bug**
`compiler.has_function_attribute('ifunc')` is returning false on clang++ (e.g. with clang 14).

**To Reproduce**
```
project('foo', ['c', 'cpp'])
c_compiler = meson.get_compiler('c')
cpp_compiler = meson.get_compiler('cpp')
c_compiler.has_function_attribute('ifunc')
cpp_compiler.has_function_attribute('ifunc')
```

Output:
```
The Meson build system
Version: 1.2.99
Source dir: /home/bmerry/tmp/ifunc-clang
Build dir: /home/bmerry/tmp/ifunc-clang/build
Build type: native build
Project name: foo
Project version: undefined
C compiler for the host machine: clang (clang 14.0.0-1ubuntu1 "Ubuntu clang version 14.0.0-1ubuntu1.1")
C linker for the host machine: clang ld.bfd 2.38
C++ compiler for the host machine: clang++ (clang 14.0.0-1ubuntu1 "Ubuntu clang version 14.0.0-1ubuntu1.1")
C++ linker for the host machine: clang++ ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports function attribute ifunc: YES 
Compiler for C++ supports function attribute ifunc: NO 
Build targets in project: 0
```

Relevant part of log:
```
Working directory:  /home/bmerry/tmp/ifunc-clang/build/meson-private/tmptv1i4cbr
Code:
 extern "C" {int my_foo(void) { return 0; }static int (*resolve_foo(void))(void) { return my_foo; }}int foo(void) __attribute__((ifunc("resolve_foo")));
-----------
Command line: `clang++ /home/bmerry/tmp/ifunc-clang/build/meson-private/tmptv1i4cbr/testfile.cpp -o /home/bmerry/tmp/ifunc-clang/build/meson-private/tmptv1i4cbr/output.obj -c -D_FILE_OFFSET_BITS=64 -O0 -fpermissive -Werror=implicit-function-declaration -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument -Werror=attributes` -> 1
stderr:
/home/bmerry/tmp/ifunc-clang/build/meson-private/tmptv1i4cbr/testfile.cpp:1:129: error: ifunc must point to a defined function
extern "C" {int my_foo(void) { return 0; }static int (*resolve_foo(void))(void) { return my_foo; }}int foo(void) __attribute__((ifunc("resolve_foo")));
                                                                                                                                ^
1 error generated.
```

It appears that clang++ doesn't like the resolver being declared static, even though clang (C compiler) is happy with it.

I will submit a PR to fix this.

**Expected behavior**
The ifunc attribute should be detected as supported.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? No
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.) Ubuntu 22.04
* what Python version are you using e.g. 3.8.0: 3.10.12
* what `meson --version`: 1.2.99 (git ae7a9b0)

--- a/mesonbuild/compilers/c_function_attributes.py
+++ b/mesonbuild/compilers/c_function_attributes.py
@@ -137,7 +137,7 @@ CXX_FUNC_ATTRIBUTES = {
     'ifunc':
         ('extern "C" {'
          'int my_foo(void) { return 0; }'
-         'static int (*resolve_foo(void))(void) { return my_foo; }'
+         'int (*resolve_foo(void))(void) { return my_foo; }'
          '}'
          'int foo(void) __attribute__((ifunc("resolve_foo")));'),
 }

